src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -224,6 +224,7 @@ public static int OffsetToStringData
         /// <returns>true if given type is reference type or value type that contains references</returns>
         [Intrinsic]
         public static bool IsReferenceOrContainsReferences<T>()
+            where T : allows ref struct
```

Comment:
I think we should omit `IsReferenceOrContainsReferences` change in this PR and open an issue to deal with it separately.

Re: https://github.com/dotnet/runtime/issues/102671#issuecomment-2138007758

Reply:
Yup, I have it updated locally to do that. Just added more tests for things. I'll open an issue as well.